File: backend-ms1/src/main/java/ma/zs/zyn/service/facade/admin/inscriptionref/InviteStudentAdminService.java
package ma.zs.zyn.service.facade.admin.inscriptionref;

import java.util.List;
import ma.zs.zyn.bean.core.inscriptionref.InviteStudent;
import ma.zs.zyn.dao.criteria.core.inscriptionref.InviteStudentCriteria;
import ma.zs.zyn.zynerator.service.IService;



public interface InviteStudentAdminService extends  IService<InviteStudent,InviteStudentCriteria>  {

    List<InviteStudent> findByEtudiantId(Long id);
    int deleteByEtudiantId(Long id);
    long countByEtudiantRef(String ref);




}

File: frontend-bqckup/src/app/shared/service/admin/inscriptionref/InviteStudentAdmin.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {InviteStudentDto} from '../../../model/inscriptionref/InviteStudent.model';
import {InviteStudentCriteria} from '../../../criteria/inscriptionref/InviteStudentCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class InviteStudentAdminService extends AbstractService<InviteStudentDto, InviteStudentCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'admin/inviteStudent/';
    }

    public constrcutDto(): InviteStudentDto {
        return new InviteStudentDto();
    }

    public constrcutCriteria(): InviteStudentCriteria {
        return new InviteStudentCriteria();
    }
}
